packages/react/src/components/UIShell/components/styles/_header.scss

```diff
@@ -5,8 +5,35 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+@use '@carbon/styles/scss/theme' as *;
+@use '@carbon/styles/scss/motion' as *;
+@use '@carbon/styles/scss/utilities/custom-property' as custom-property;
+
 $prefix: 'cds' !default;
 
 .#{$prefix}--header__action:focus {
   border-width: 2px;
 }
+
+// TODO: remove when https://github.com/carbon-design-system/carbon/pull/18725 is released
+.#{$prefix}--header-action .#{$prefix}--tooltip-content {
+  color: $text-primary;
+}
+
+.#{$prefix}--header-action.#{$prefix}--popover--open
+  .#{$prefix}--header-action__button {
+  background-color: $layer;
+}
+
+.#{$prefix}--header-action .#{$prefix}--toggletip-content {
+  @include custom-property.declaration('link-text-color', $link-primary);
+  @include custom-property.declaration(
+    'link-hover-text-color',
+    $link-primary-hover
+  );
+  @include custom-property.declaration(
+    'link-visited-text-color',
+    $link-visited
+  );
+  @include custom-property.declaration('link-focus-text-color', $focus);
+}
```

Comment:
```
@include custom-property.declaration('button-focus-color', theme.$focus);
  @include custom-property.declaration('link-focus-text-color', theme.$focus);
```

I think we need to add these two tokens also to make sure we're swapping back everything added here: https://github.com/carbon-design-system/carbon/blob/main/packages/styles/scss/components/toggletip/_toggletip.scss#L53-L74